tests/validators/routing/test_types.py

```diff
@@ -98,6 +98,19 @@ def test_resolve_grand_calculated_summary_value_source_json_type():
     )
 
 
+def test_resolve_metadata_summary_value_source_json_type():
```

Comment:
I'd parametrize this test too, and do a case for each of the different possible metadata json types